core/variant/variant.h

```diff
@@ -91,11 +94,14 @@ class Variant {
 		VECTOR3,
 		VECTOR3I,
 		TRANSFORM2D,
+		VECTOR4,
+		VECTOR4I,
```

Comment:
I know it's arbitrary, but this order doesn't really make sense. Why should Transform2D go in-between Vector3 and Vector4? I think we should consider reordering the types in the enum, maybe logical grouping, maybe alphabetical.

For local grouping it would make sense to have Vector2/3/4 first, then Rect2/AABB, then Plane/Quat(/Color?), then Transform2D, Basis, Transform3D, Projection.

Reply:
Its together with the other 4 component types so to me it makes sense that way. Moving then is far too much work also, so won't do without a very good reason.

Reply:
Upon further inspection, I think the problem is that TRANSFORM2D should go above VECTOR3. We could change this in a separate PR I guess.

Reply:
Also keep in mind, while I guess this isn't a problem since we're still in Alpha and we accept breakage, that adding new types in the middle will segfault any existing GDExtension that assumes the old numbering.